src/include/txn_inline.h

```diff
@@ -291,20 +295,31 @@ __wt_txn_op_delete_commit_apply_timestamps(WT_SESSION_IMPL *session, WT_REF *ref
 {
     WT_TXN *txn;
     WT_UPDATE **updp;
+    wt_timestamp_t *timestamp;
     uint8_t previous_state;
 
     txn = session->txn;
 
+    /* Lock the ref to ensure we don't race with page instantiation. */
+    WT_REF_LOCK(session, ref, &previous_state);
+
     /*
-     * Lock the ref to ensure we don't race with eviction freeing the page deleted update list or
-     * with a page instantiate.
+     * Timestamps are in the page deleted structure for truncates, or in the updates in the case of
+     * instantiated pages. Both commit and durable timestamps need to be updated.
      */
-    WT_REF_LOCK(session, ref, &previous_state);
+    if (previous_state == WT_REF_DELETED) {
+        timestamp = &ref->ft_info.del->timestamp;
```

Comment:
Can we stop using `timestamp` I feel like it doesn't improve readability, is there another reason for it?

Reply:
I could easily be wrong, but I think the timestamp isn't necessarily one kind of timestamp, so it's a generic timestamp -- if that's wrong, I'd strongly support something more descriptive. What did you have in mind?

Reply:
Something along the lines of:
```
    if (previous_state == WT_REF_DELETED) {
        if (ref->ft_info.del->timestamp == WT_TS_NONE) {
            ref->ft_info.del->timestamp = txn->commit_timestamp;
            ref->ft_info.del->durable_timestamp = txn->durable_timestamp;
        }
    }
```
Then for the similar code below:
```
    else {
            /*
             * The timestamp is in the update for operations other than truncate. Both commit and
             * durable timestamps need to be updated.
             */
            upd = op->u.op_upd;
            if (upd->start_ts == WT_TS_NONE) {
                upd->start_ts = txn->commit_timestamp;
                upd->durable_ts = txn->durable_timestamp;
            }
        }
```

This has also me wonder if we should just rename timestamp to commit_timestamp in the del structure?

Reply:
Honestly, I have no opinion, I don't really know how that stuff works.

Reply:
Still though does the timestamp change seem reasonable? I just don't like the idea of pulling it out into its own variable.

Reply:
Oh, sorry, @luke-pearson -- I finally figured out what you meant.

Yeah, I totally agree, I think it's not worth doing at best and confusing at worst. I thought the same thing when I was in there, but was feeling sufficiently badly about changing tons of code around fast-truncate that I left it alone.

Fix pushed in 38a96f7a9 for your review.

Reply:
Much better, thanks. 👍